hazelcast/src/main/java/com/hazelcast/map/impl/BasicMapContextQuerySupport.java

```diff
@@ -142,9 +142,10 @@ public Set queryLocalMember(String mapName, Predicate predicate, IterationType i
             if (partitionIds.isEmpty()) {
                 return result;
             }
-        } catch (QueryResultSizeExceededException e) {
-            throw ExceptionUtil.rethrow(e);
         } catch (Throwable t) {
+            if (t.getCause() instanceof QueryResultSizeExceededException) {
```

Comment:
My suggestion does not change the behavior but only changes where we handle to getCause thingy. My suggestion is that let the getCause done in `BasicMapContextQuerySupport.addResultsOfPagingPredicate ` , `BasicMapContextQuerySupport.addResultsOfPredicate ` and `TransactionalMapProxySupport . collectResults ` as it is done in `BasicMapContextQuerySupport.queryOnPartitions ` via ExceptionUtil.rethrow . Then  you can revert the  catch part that you changed in this pr. This way, these methods will either throw QueryResultSizeExceededException(which will be catched and rethrown) or any other runtime exception which will be logged.